extensions/counter/src/main/java/org/radargun/stages/CounterTestStage.java

```diff
@@ -0,0 +1,132 @@
+         return new RatioOperationSelector.Builder().add(CounterOperations.DECREMENT_AND_GET, 1).build();
+      } else if (operationName.equals(OperationName.ADD_AND_GET)) {
+         return new RatioOperationSelector.Builder().add(CounterOperations.ADD_AND_GET, 1).build();
+      } else if (operationName.equals(OperationName.COMPARE_AND_SET)) {
+         return new RatioOperationSelector.Builder().add(CounterOperations.COMPARE_AND_SET, 1).build();
+      } else {
+         throw new IllegalArgumentException("Unknown operation!");
+      }
+   }
+
+   @Override
+   public OperationLogic getLogic() {
+      return new CounterLogic();
+   }
+
+   /**
+    *
+    */
+   protected class CounterLogic extends OperationLogic {
+      private CounterOperations.Counter counter;
+      private long previousValue;
+
+      public CounterLogic() {
+         this.previousValue = initialValue;
+      }
+
+      @Override
+      public void init(Stressor stressor) {
+         super.init(stressor);
+         this.counter = counterOperations.getCounter(counterName);
+         log.warn("Transactions ignored for Counter operations!");
+         stressor.setUseTransactions(false);//transactions for counter do not make sense
+      }
+
+      @Override
+      public void run(Operation operation) throws RequestException {
+         if (operation == CounterOperations.INCREMENT_AND_GET) {
+            Invocation<Long> invocation = new CounterInvocations.IncrementAndGet(counter);
+            long currentValue = stressor.makeRequest(invocation);
+            if (currentValue == previousValue) {
+               throw new IllegalStateException("Inconsistent counter! Expected greater than " + previousValue);
+            } else {
+               previousValue = currentValue;
+            }
+         } else if (operation == CounterOperations.DECREMENT_AND_GET) {
+            Invocation<Long> invocation = new CounterInvocations.DecrementAndGet(counter);
+            long currentValue = stressor.makeRequest(invocation);
+            if (currentValue == previousValue) {
+               throw new IllegalStateException("Inconsistent counter! Expected lesser than " + previousValue);
+            } else {
+               previousValue = currentValue;
+            }
+         } else if (operation == CounterOperations.ADD_AND_GET) {
+            Invocation<Long> invocation = new CounterInvocations.AddAndGet(counter, delta);
+            long currentValue = stressor.makeRequest(invocation);
+            if (currentValue == previousValue) {
+               throw new IllegalStateException("Inconsistent counter! Expected value different from previous.");
+            } else {
+               previousValue = currentValue;
+            }
+         } else if (operation == CounterOperations.COMPARE_AND_SET) {
+            //This operation is testable only with a single thread. Running with multiple threads would cause
```

Comment:
IMO a failed invocation is valid as well, just don't check that success is always true. Implementation might allow spurious failures.

Reply:
You could record success rate and add that to the results (user can then verify that success rate is 1/numThreads).